### ISSUE TITLE
How to convert Kinv(or Kmat) into a (FoVx,FoVy,H,W) format?[Bug/issue]

### ISSUE THREAD

**Issue by zz7379** [lab4d-org/lab4d#36]
Hi, gengshan. Thanks for your excellent code.
Could you share how to convert Kinv(or Kmat) into a (FoVx,FoVy,H,W) format?
I notice that if 
`            FoVx = 2 * torch.arctan(Kmat[0, 2] / Kmat[0, 0])
            FoVy = 2 * torch.arctan(Kmat[1, 2] / Kmat[1, 1])`
It is not correct. 
When evaluating, the true resolution(e.g. 64x64) doesn't match [Kmat[0, 2], Kmat[1, 2]] (e.g. [60.01, 68.03])

**Comment by gengshan-y** [comment]
Hi,

`Kmat` is in the following format
```
[[focal_x, 0, px],
 [0, focal_y, py],
 [0, 0,       1]]
```
where focal_x/y and px/y are in pixel unit


To convert focal to fov, you need
```
focal_unit = 2 * focal / img_size # normalize
fov = 2 * torch.atan(1 / focal_unit)
```
hope it helps.

**Comment by zz7379** [comment]
Thanks for your reply!
Actually, i find:
           `# wrong
            FoVx = 2 * torch.arctan(Kmat[0, 2] / Kmat[0, 0])
            FoVy = 2 * torch.arctan(Kmat[1, 2] / Kmat[1, 1])` 

            `# correct
            hxy = torch.zeros([4, 3]).cuda()
            hxy[0]=torch.tensor([0,0,1])
            hxy[1]=torch.tensor([0,W,1])
            hxy[2]=torch.tensor([H,0,1])
            hxy[3]=torch.tensor([H,W,1])
            dir = hxy @ Kinv.permute(1, 0)
            xyrange = dir.max(axis=0)[0] - dir.min(axis=0)[0]
            FoVx = 2 * torch.atan(xyrange[0] / 2)
            FoVy = 2 * torch.atan(xyrange[1] / 2)`

Suitable for OpenGL style camera!
